### ISSUE TITLE
Finite-element frequency domain

### ISSUE THREAD

**Issue by simbilod** [HelgeGehring/femwell#46]
For FEFD, we had "issues" due to source directionality

Is this even a problem? In https://nbviewer.org/github/fancompute/workshop-invdesign/blob/master/01_First_simulation.ipynb they just compute the Poynting vector somewhere away from the source. 

Could we simply compare the amount of propagating energy and/or the fields close to the source vs at the target outputs modes/positions to get the S-parameters?

@HelgeGehring

**Comment by HelgeGehring** [comment]
With the current way, we just define the field in a certain location. So to be very correct, it's not a source and thus there's no directionality.
I'd guess that's okay, as we could also put it close to a PML and just absorb whatever goes in the other direction.
We could measure the power in the modes at the ports by calculating overlap integrals, right?

The main problem I have at the moment in my understanding is how we would calculate back reflection. but maybe we could solve that by calculating an overlap integral before and after the position where we set the field?

**Comment by simbilod** [comment]
yes I think you can look at the Poynting vector going toward the source from the simulation, on the side of the simulation

**Comment by simbilod** [comment]
could we somehow "normalize" the source/input field by computing its flux near the position where the source/field is set in a simple waveguide geometry, and attribute deviations from that to reflections?

**Comment by simbilod** [comment]
it looks like it's a little more subtle http://hade.ch/docs/report_FDFD.pdf

**Comment by timurdogan** [comment]
Few inputs here, if you find the correct eigenmode with all full vectorial components, Ex,Ey,Hx,Hy, and launch with those, there are no theoretical reflections and directivity comes naturally. The eigenmode calculation error can result in mismatch. However, there is also a limit on any FDTD/FDFD code based on settings used like acceleration parameters and 16bit/32bit/64bit processing and scaling to not run out of bits. This can be tested by placing a straight single-mode waveguide and running it and simulating loss vs. waveguide length, looking at absorption in PML boundaries vs. what is launched. The goal should be within an error of better than -40dB in transmission and reflection. If you want to learn about these limits, I suggest this book;
_"Computational Electrodynamics: The Finite-Difference Time-Domain Method, Alan Taflove"_

**Comment by HelgeGehring** [comment]
thanks @timurdogan ! looks like a great ressource! I'll start with the waveguide example

I've done first steps in #59, mostly docs and a simple waveguide example, but more to come

**Comment by timurdogan** [comment]
Thank you!
